packages/flutter/lib/src/services/system_navigator.dart

```diff
@@ -67,20 +67,34 @@ class SystemNavigator {
 
   /// Notifies the platform for a route information change.
   ///
-  /// On web, creates a new browser history entry and update URL with the route
-  /// information. Whether the history holds one entry or multiple entries is
-  /// determined by [selectSingleEntryHistory] and [selectMultiEntryHistory].
+  /// On web, this method behaves differently based on the single-entry or
+  /// multiple-entries history mode. Use the [selectSingleEntryHistory] and
+  /// [selectMultiEntryHistory] to toggle between modes.
   ///
-  /// Currently, this is ignored on other platforms.
+  /// For single-entry mode, this method replaces the current url and state in
+  /// the current history entry. The flag `replace` is ignored.
+  ///
+  /// For multiple-entries mode, this method creates a new history entry on top
```

Comment:
is this actually right? I would have expected that when replace is false, a new entry is created and put on top. If it is true, the new entry replaces the existing one?

Reply:
its a typo, fixed